client/api/src/main/java/io/smallrye/graphql/client/Response.java

```diff
@@ -17,4 +18,6 @@
     boolean hasData();
 
     boolean hasError();
+
+    List<Map.Entry<String, String>> getHeaders();
```

Comment:
Why not simply `Map<String, String>`?  This is unnecessarily complicated.

Reply:
Headers can be repeated, so a `Map` won't work. In JAX-RS there's `MultivaluedMap` which extends `Map<K, List<V>>`. In vert.x there's `MultiMap` which implements `Iterable<Entry<String, String>>`.

Reply:
Hmm okay, I didn't realise that. We certainly want to avoid depending on JAX-RS or Vert.x classes, so...  `List<Map.Entry<String, String>>` makes retrieving a header by its name somewhat complicated, so maybe we should automatically transform the headers to a `Map<K, List<V>>`, where it is more straightforward?

Reply:
Rebase done.
I changed to HeadersMultiMap as suggested, I'm also filling the MultiMap in the constructor.

Reply:
@AnakinPt Ah, no no, I meant we want to avoid depending on Vert.x in the common module, you've just added that dependency. `Response` should store a `Map<String, List<String>>`, not a `MultiMap`.

Reply:
Sorry. I misunderstood. Later on today I'll revert and do the changes proposed.

Reply:
Changes done. I think now we can work with this.